sdk/identity/Azure.Identity/src/CredentialPipeline.cs

```diff
@@ -69,5 +70,13 @@ public CredentialPipeline WithAuthorityHost(Uri authorityHost)
         {
             return new CredentialPipeline(authorityHost, HttpPipeline, Diagnostics);
         }
+
+        private class CredentialResponseClassifier : ResponseClassifier
+        {
+            public override bool IsRetriableResponse(HttpMessage message)
+            {
+                return base.IsRetriableResponse(message) || message.Response.Status == 404;
```

Comment:
Interesting, would retrying multiple times in a row help when server returns 404?

Reply:
Yes apparently the service returns 404 when it is upgrading. https://docs.microsoft.com/en-us/azure/active-directory/managed-identities-azure-resources/how-to-use-vm-token#error-handling

Reply:
🤦‍♂